src/ufuncs.c

```diff
@@ -2455,6 +2455,14 @@ static void points_func(char** args, npy_intp* dimensions, npy_intp* steps, void
   GEOSCoordSequence* coord_seq = NULL;
   GEOSGeometry** geom_arr;
 
+  // check the ordinate dimension before calling GEOSCoordSeq_create_r
+  if (dimensions[1] < 2 || dimensions[1] > 3) {
+    PyErr_Format(PyExc_ValueError,
+                 "The ordinate (last) dimension should be 2 or 3, got %ld",
+                 dimensions[1]);
+    return;
```

Comment:
Currently we are just passing through whatever dimension we get through to GEOS. But `GEOSCoordSeq_create_r` is documented to only accept dimension 2 and 3. And so when now passing dimension 4, we actually get a wrong result instead of an error (with previous versions of GEOS). See https://github.com/libgeos/geos/pull/721#discussion_r1038752015

And even if GEOS would change that to raise an error again, checking this on our side is better I think. It is consistent with other creation ufuncs (eg for `linestrings()` we have this check on our side as well), and it avoids that you can accidentally create points for 1D coords (`shapely.points([1])` currently creates POINT (1 0), so filling in with zero; with this change that will raise an error).